### ISSUE TITLE
Qwen-72B-chat-GPTQ TP=4 ERROR

### ISSUE THREAD

**Issue by Hukongtao** [NVIDIA/TensorRT-LLM#1344]
### System Info

xx

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

xx

### Expected behavior

xx

### actual behavior

xx

### additional notes

xx

**Comment by Hukongtao** [comment]
Do you test your code before releasing it?

**Comment by byshiue** [comment]
Please fill the info to help reproduce the issue. We have tests before releasing.

**Comment by Hukongtao** [comment]
> Please fill the info to help reproduce the issue. We have tests before releasing.

I followed the official documentation:
https://github.com/NVIDIA/TensorRT-LLM/blob/main/examples/qwen/README.md#int4-gptq
![image](https://github.com/NVIDIA/TensorRT-LLM/assets/16505966/b818e1e6-5d17-4ab3-bc6e-82c7004f7637)
I encountered the first error when converting the model weights:
```python
python3 examples/qwen/convert_checkpoint.py \
    --model_dir  ./Qwen-72B-Chat-Int4/ \
    --output_dir ./Qwen-72B-Chat-Int4-TRT/tllm_checkpoint_4gpu_tp4_gptq/ \
    --dtype float16 \
    --use_weight_only \
    --weight_only_precision int4_gptq \
    --tp_size 4 \
    --pp_size 1 \
    --per_group
```
![image](https://github.com/NVIDIA/TensorRT-LLM/assets/16505966/8f84288e-24e4-454e-9a24-61d17463b9d5)
```shell
ValueError: You are trying to save a non contiguous tensor: `transformer.layers.0.mlp.gate.weights_scaling_factor` which is not allowed. It either means you are trying to save tensors which are reference of each other in which case it's recommended to save only the full tensors, and reslice at load time, or simply call `.contiguous()` on your tensor to pack it before saving.
```
I can fix this error by following the modifications below:
![image](https://github.com/NVIDIA/TensorRT-LLM/assets/16505966/cb3bbbf3-19a4-4e67-a4cd-d9e0b22aee09)
https://github.com/NVIDIA/TensorRT-LLM/blob/66ca3378c61efa3154ed34a48cfc362351405eef/examples/qwen/convert_checkpoint.py#L250-L254

Then, I try to build the engine:
```python
trtllm-build \
    --checkpoint_dir   ./Qwen-72B-Chat-Int4-TRT/tllm_checkpoint_4gpu_tp4_gptq/ \
    --output_dir       ./Qwen-72B-Chat-Int4-TRT/trt_engines/int4-gptq/4-gpu/ \
    --max_batch_size   1    \
    --max_input_len    2048 \
    --max_output_len   512  \
    --gather_all_token_logits \
    --gemm_plugin float16 \
    --tp_size 4
```
Then I got:
![image](https://github.com/NVIDIA/TensorRT-LLM/assets/16505966/1ccd2265-811a-449e-bdf6-7ee24d928871)
```shell
RuntimeError: Encounter error 'The value updated is not the same shape as the original. Updated: (8192, 6144), original: (8192, 1536)' for parameter 'transformer.layers.0.attention.qkv.weight'
```
I don't know how to fix it

**Comment by Hukongtao** [comment]
One more point to add，
under commit 4bb65f216f2f1dbddb022f4fdd8925c2856baa58, it can run successfully.

**Comment by Hukongtao** [comment]
There is another problem. The Qwen model does not seem to support the `--gather_all_token_logits` parameter。
```python3

trtllm-build \
    --checkpoint_dir ./Qwen-VL-Chat-TRT/tllm_checkpoint_1gpu/ \
    --gemm_plugin float16 \
    --gpt_attention_plugin float16 \
    --lookup_plugin float16 \
    --max_input_len 2048 \
    --max_output_len 1024 \
    --max_batch_size 8 \
    --max_prompt_embedding_table_size 2048 \
    --remove_input_padding enable \
    --gather_all_token_logits \
    --output_dir ./Qwen-VL-Chat-TRT/trt_engines/Qwen-VL-7B-Chat

```
When infer, I got:
```shell
RuntimeError: Tensor names in engine are not the same as expected, to use this GenerationSession, you need to use PretrainedModel.prepare_inputs to create TRT Network inputs.
```

**Comment by Hukongtao** [comment]
Looking forward to your reply @byshiue

**Comment by adamydwang** [comment]
I suffered this problem too.
qwen-72b-chat, tp=4, smoothquant

**Comment by ZhangJinxin1** [comment]
I suffered this problem too.
qwen-72b-chat, tp=8, smoothquant

**Comment by HermitSun** [comment]
+1
qwen-72b-chat, tp=4, gptq

**Comment by Hukongtao** [comment]
> +1 qwen-72b-chat, tp=4, gptq

Is the error message you encountered the same as mine?

**Comment by HermitSun** [comment]
> > +1 qwen-72b-chat, tp=4, gptq
> 
> Is the error message you encountered the same as mine?

Yes. And I used same cmds to build the engine under trt-llm 0.9.0.dev2024040200.

**Comment by Hukongtao** [comment]
@byshiue @Tracin Do you have any plan to fix this bug?

**Comment by Hukongtao** [comment]
![image](https://github.com/NVIDIA/TensorRT-LLM/assets/16505966/e2f5d15a-0329-4cb4-acb5-931e396b3956)
If I modify the code like this, the code run successfully. But the inference result`(run.py)`  is weried.

![image](https://github.com/NVIDIA/TensorRT-LLM/assets/16505966/0f75a52b-604a-4091-acdd-d45c31c2c4b5)

**Comment by Tracin** [comment]
@Hukongtao Sure, I am working on this, I will keep you posted.

**Comment by Tracin** [comment]
@Hukongtao @HermitSun 
I will push a MR to fix this. If you want to fix it in advance, please try to replace [this](https://github.com/NVIDIA/TensorRT-LLM/blob/v0.9.0/tensorrt_llm/models/qwen/weight.py#L138-L140) section to:
```
        for suf in suffixs:
            qkv_part = model_params[prefix + "attn.c_attn." + suf]
            split_dim = qkv_part.shape[1] // 3
            q_part = torch_split(qkv_part[:, :split_dim], dim=1)
            k_part = torch_split(qkv_part[:, split_dim : split_dim * 2], dim=1)
            v_part = torch_split(qkv_part[:, -split_dim:], dim=1)
            qkv_part = torch.cat([q_part, k_part, v_part], 1)
            qkv_weight_list.append(qkv_part)
```

**Comment by Tracin** [comment]
@adamydwang @ZhangJinxin1
As for SmoothQuant problem, do we have an issues on Github? Let's move to there.

**Comment by HermitSun** [comment]
> @Hukongtao @HermitSun I will push a MR to fix this. If you want to fix it in advance, please try to replace [this](https://github.com/NVIDIA/TensorRT-LLM/blob/v0.9.0/tensorrt_llm/models/qwen/weight.py#L138-L140) section to:
> 
> ```
>         for suf in suffixs:
>             qkv_part = model_params[prefix + "attn.c_attn." + suf]
>             split_dim = qkv_part.shape[1] // 3
>             q_part = torch_split(qkv_part[:, :split_dim], dim=1)
>             k_part = torch_split(qkv_part[:, split_dim : split_dim * 2], dim=1)
>             v_part = torch_split(qkv_part[:, -split_dim:], dim=1)
>             qkv_part = torch.cat([q_part, k_part, v_part], 1)
>             qkv_weight_list.append(qkv_part)
> ```

It works in my situation. Thank you for your effort!

**Comment by Hukongtao** [comment]
> @Hukongtao @HermitSun I will push a MR to fix this. If you want to fix it in advance, please try to replace [this](https://github.com/NVIDIA/TensorRT-LLM/blob/v0.9.0/tensorrt_llm/models/qwen/weight.py#L138-L140) section to:
> 
> ```
>         for suf in suffixs:
>             qkv_part = model_params[prefix + "attn.c_attn." + suf]
>             split_dim = qkv_part.shape[1] // 3
>             q_part = torch_split(qkv_part[:, :split_dim], dim=1)
>             k_part = torch_split(qkv_part[:, split_dim : split_dim * 2], dim=1)
>             v_part = torch_split(qkv_part[:, -split_dim:], dim=1)
>             qkv_part = torch.cat([q_part, k_part, v_part], 1)
>             qkv_weight_list.append(qkv_part)
> ```

Thank you for your work!
